docs/contributing/building.rst

```diff
@@ -135,6 +135,63 @@ or source it into your shell,
 
    . $BUILD_DIR/python_env.sh
 
+Resolving Common Build Issues
+=============================
+
+If you have having issues from a clean build directory (i.e., empty directory),
+
+1. Make sure you have also checked out any git submodules: ``git submodule
+   update --recursive --init``
+
+2. If you are using Conda, make sure that you have activated your environment
+   for both the ``cmake`` and ``make`` steps: ``conda activate katana-dev``
+
+If you were previously successful building but now you are seeing ``cmake`` or
+unexpected build errors after updating your source directory,
+
+1. Make sure you have also checked out any git submodules: ``git submodule
+   update --recursive --init``
+
+2. Check if there were any system build environment changes since the last time
+   you successfully built. To update your environment, run
+   ``scripts/setup_dev_ubuntu.sh``.
+
+   This requires root privileges, if you don't have root, it is likely that
+   your system administrator has already updated your build environment.
```

Comment:
"If you are not using Conda,"

Reply:
Also 2 seems to be the same as 3 in many ways. Should they be merged?

Reply:
> "If you are not using Conda,"

`setup_dev_ubuntu.sh` uses `sudo`:
https://github.com/ddn0/katana/blob/master/scripts/setup_dev_ubuntu.sh#L18

Reply:
I meant that you should not be instructed to run `setup_dev_ubuntu.sh` at all if you are using conda. It's not needed.